### ISSUE TITLE
Rule request: no identical branches

### ISSUE THREAD

**Issue by lo1tuma** [realm/SwiftLint#4665]
### New rule request

When refactoring or deleting code I sometimes end up with an if-else where both branches have the same code. It would be nice if SwiftLint could report those branches so I can get rid of the code branching completely. Ideally this could be also applied to duplicated branches in switch statements.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Having two branches that execute the same code based on a different condition means that either the condition is not necessary at all (e.g. for an `if-else` statement) or it could be refactored to combine the duplicated branches (e.g. when having a more complicated if-elseif-else statement). See also the [sonarjs rule for JavaScript](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/no-duplicated-branches.md).

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Violation:

```swift
if foo {
  bar()
} else {
  bar() // duplicated
}

if foo {
  bar()
else if baz {
  bar() // duplicated
} else
  other()
}

switch foo {
case "a":
  bar()
case "b":
  bar() // duplicated
}
```

Non-Violation:

```swift
if foo {
  bar()
} else {
  other()
}

if foo || bar {
  bar()
} else
  other()
}

switch foo {
case "a", "b":
  bar()
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

No, it doesn’t need to be configurable

6. Should the rule be opt-in or enabled by default? Why?

I think it would be good as a default rule because this rule almost always detects bad structured code. But personally I don’t really care about this, because I’m not relying on any defaults, using `only_rules` instead.
